templates/etc/marketplace.html

```diff
@@ -0,0 +1,115 @@
+          <input id="commission-max-price" name="max" type="text" class="input" value="${form.max}">
+        </div>
+      </div>
+
+      <label for="commish-currency">Display Currency</label>
+      <select class="input last-input" name="currency" id="commish-currency">
+        $for char, info in currencies.iteritems():
+          <option value="${char}" $:{selected(form.currency == char)}>${info.name}</option>
+      </select>
+
+      <button class="button">Search</button>
+      <p class="color-lighter tags-help"><i><a href="/help/marketplace" target="_blank">Marketplace search help</a></i></p>
+    </form>
+  </div>
+
+  <div class="sectioned-main content">
+    <h3>Artist Results</h3>
+
+    $if results:
+      $for artist in results:
+        <div class="clear marketplace-result">
+          <a href="/~${LOGIN(artist['username'])}" class="avatar">
+            $ avatar = artist['user_media']['avatar'][0]
+            <img src="${avatar['display_url']}" alt="" />
+          </a>
+          $# artist's commission info
+          <div class="artist-links">
+            <a href="/~${artist['username']}#user-commissions" class="username">${artist['username']}</a>
+            <a href="/submissions/${artist['username']}">[Gallery]</a>
+            $if artist['searchquery']:
+              <a href="/search?${artist['searchquery']}">[Tagged: ${artist['examplecount']}]</a>
+          </div>
+          $# commission price range
+          <div class="artist-prices">
+            $if artist['pricemin']:
+              $# display low end of price range in artist's currency
+              <span>$:{'from ' if artist['pricemax'] else ''}${SYMBOL(artist['pricesettings'])} ${PRICE(artist['pricemin'])}</span>
+              $if artist['pricemax'] and artist['pricemax'] != artist['pricemin']:
+                $# display high end of price range in artist's currency
+                <span>to ${SYMBOL(artist['pricesettings'])} ${PRICE(artist['pricemax'])}</span>
+              $if artist['pricesettings'] != form.currency and artist['localmin']:
+                $if artist['localmax'] and artist['localmax'] != artist['localmin']:
+                  $# display converted prices (range)
+                  (<span>Approx. ${SYMBOL(form.currency)} ${PRICE(artist['localmin'])}</span>
+                  <span>to ${SYMBOL(form.currency)} ${PRICE(artist['localmax'])}</span>)
+                $else:
+                  $# display converted price (single)
+                  (<span>Approx. ${SYMBOL(form.currency)} ${PRICE(artist['localmin'])}</span>)
+            $else:
+              Price not declared
+          </div>
+          $# matching commission classes
+          $if form.c or form.pc:
+            <div class="artist-classes">
+              <strong>Matched: ${artist['class']}</strong>
+            </div>
+          $# commission description
+          <div class="formatted-content marketplace-desc-preview">
+            $if artist.get('description'):
```

Comment:
If the user sets up their commission prices but does not fill out the "Additional Information" field, the value may be None. If they fill it out then erase it at a later date, it may be an empty string. But in either case, I can't see a situation where the key itself will be missing under the current setup.